test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java

```diff
@@ -0,0 +1,276 @@
+import java.awt.HeadlessException;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UIManager.LookAndFeelInfo;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import static javax.swing.UIManager.getInstalledLookAndFeels;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4390885
+ * @summary This test checks CCC #4390885, which verifies that it should be
+ *          possible to set the location of the JFileChooser.
+ * @run main JFileChooserSetLocationTest
+ */
+public class JFileChooserSetLocationTest {
+
+    public static final String SHOW_DIALOG_OUTSIDE_THE_PANEL =
+            "ShowFileChooser OUTSIDE the Panel";
+    public static final String SHOW_DIALOG_OVER_THE_PANEL =
+            "ShowFileChooser OVER the Panel";
+    public static final String SHOW_SAVE_DIALOG_OVER_THE_PANEL =
+            "ShowSaveDialog";
+    private static final int TOLERANCE_LEVEL = 6;
+    private static final int xOut = 75;
+    private static final int yOut = 75;
+    private static Robot robot;
+    private static JPanel panel;
+    private static MyFileChooser fileChooser;
+    private static int xIn;
+    private static int yIn;
+    private static JButton btn;
+    private static JButton btn1;
+    private static JButton btn2;
+    private static JFrame frame;
+
+    public static void main(String[] s) throws Exception {
+        robot = new Robot();
+        robot.setAutoWaitForIdle(true);
+        robot.setAutoDelay(200);
+
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            try {
+                AtomicBoolean lafSetSuccess = new AtomicBoolean(false);
+                SwingUtilities.invokeAndWait(() -> {
+                    lafSetSuccess.set(setLookAndFeel(laf));
+                    if (lafSetSuccess.get()) {
+                        createUI();
+                    }
+                });
+                if (!lafSetSuccess.get()) {
+                    continue;
+                }
+                robot.waitForIdle();
+
+                AtomicReference<Point> pt = new AtomicReference<>();
+                SwingUtilities.invokeAndWait(
+                        () -> pt.set(panel.getLocationOnScreen()));
+                Point panel_loc = pt.get();
+                xIn = (panel_loc.x + panel.getSize().width) / 2;
```

Comment:
The getSize should be called on EDT as well.

Reply:
Changed.